doc/overviews/helm-charts-release-process.md

```diff
@@ -0,0 +1,237 @@
+    GitHub->>GitHub: Create GitHub Release with changelog
+
+    Note over Dev, ArtifactHub: ⚙️ Phase 3: Helm Chart Publishing
+    GitHub->>HelmWF: Release Published Event triggers workflow
+    HelmWF->>HelmWF: Configure GPG key for signing
+    HelmWF->>HelmWF: Build chart dependencies<br/>(make helm-dependency-build)
+    HelmWF->>HelmWF: Package & sign pre-built chart<br/>(make helm-package-sign)
+    HelmWF->>GitHub: Upload chart assets<br/>(.tgz + .prov signature files)
+    HelmWF->>ChartRepo: GitHub API repository_dispatch<br/>event_type: "chart-created"
+
+    Note over Dev, ArtifactHub: 🔄 Phase 4: Helm Charts Repository Processing
+    ChartRepo->>SyncChart: Trigger sync-chart-created.yaml workflow
+    SyncChart->>GitHub: Download chart from release asset URL
+    SyncChart->>SyncChart: Update charts/index.yaml<br/>Add new chart version
+    SyncChart->>ChartRepo: Commit updated index to main branch
+    ChartRepo->>SyncIndex: Trigger sync-index.yaml workflow<br/>(on charts/index.yaml change)
+    SyncIndex->>SyncIndex: Copy index.yaml to gh-pages branch
+    SyncIndex->>ChartRepo: Publish to kuadrant.io/helm-charts
+
+    Note over Dev, ArtifactHub: 🎯 Phase 5: Artifact Hub Publishing
+    ChartRepo->>ArtifactHub: Updated helm repository detected
+    ArtifactHub->>ArtifactHub: Auto-discovery scan of repository
+    ArtifactHub->>ArtifactHub: Chart available at artifacthub.io
+```
+
+## Detailed Process
+
+### Phase 1: Release Preparation
+
+**Trigger**: Developer manually runs the [Automated Release Workflow](../../.github/workflows/automated-release.yaml)
+
+**Steps**:
+1. **Update Versions**: The workflow updates `release.yaml` with all component versions:
+    - Kuadrant Operator version
+    - Authorino Operator version
+    - Limitador Operator version
+    - DNS Operator version
+    - WASM Shim version
+    - Console Plugin version
+
+2. **Run Prepare Release**: Executes `make prepare-release` which:
+    - Runs the [release script](../../utils/release/release.sh) in phases:
+        - `pre-validation` - Validates all dependencies are available
+        - `dependencies` - Processes and validates dependency versions
+        - `operator` - Builds operator manifests and helm charts
+        - `post-validation` - Final validations
+    - **Builds Helm Charts** via `make_chart.sh` [script](../../utils/release/operator/make_chart.sh) which:
+        - Sets the desired WASM shim image in manager deployment
+        - Sets the operator image reference
+        - Builds helm chart templates from kustomize manifests (`kustomize build config/helm`)
+        - Updates Chart.yaml with version and dependency versions
+
+3. **Create Pull Request**:
+    - Creates base branch `release-v{MAJOR.MINOR}` (e.g., `release-v1.0`) if it doesn't exist
+    - Creates working branch `release-v{FULL_VERSION}` (e.g., `release-v1.0.0`)
+    - Opens PR from working branch to base branch
+
+### Phase 2: Release Creation
```

Comment:
We could link the [RELEASE.md document](https://github.com/Kuadrant/kuadrant-operator/blob/main/RELEASE.md) and here briefly explain that one can chose to trigger the _Automated Release_ workflow or follow the manual steps.